EspressoSystems/HotShot
# [DEPENDENCY_REFACTOR] - Integrate upgradability into dependency tasks

Closes #3356.

### This PR: 
* Integrates the formed and decided upgrade certificates into dependency tasks.
* Renames functions that take non-`Arc<RwLock<...>>` decided upgrade cert as parameter to `temp_*`, and adds new functions that take `Arc<RwLock<...>>`.
  * The dependency tasks now all use the new functions, but the old consensus task uses `temp_*` functions that will be replaced in a separate PR.

### This PR does not: 
* Make improvements to the existing upgradability code, which will be done later. Relevant issues:
  * #3378
  * #3357

### Key places to review: 
* Whether upgrade certificates are added correctly in the dependency tasks.
* Whether any update is missed.
* (`temp_*` functions may be ignored since they only have the name change.)

### How to test this PR:
* Run `RUST_LOG=error,hotshot_task_impls=info just cargo run --example all-push-cdn --features example-upgrade --features dependency-tasks -- --config_file ./crates/orchestrator/run-config.toml`.
  * Note the `dependency-tasks` feature.
* The upgrade should occur for view 25.
